api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2195,6 +2195,15 @@ service AuthService {
   // DeleteAllDatabaseServers removes all registered database proxy servers.
   rpc DeleteAllDatabaseServers(DeleteAllDatabaseServersRequest) returns (google.protobuf.Empty);
 
+  // GetAllDatabaseServices returns all DatabaseService resources.
+  rpc GetAllDatabaseServices(google.protobuf.Empty) returns (types.DatabaseServiceV1List);
+  // UpsertDatabaseService registers a new DatabaseService.
+  rpc UpsertDatabaseService(types.UpsertDatabaseServiceRequest) returns (google.protobuf.Empty);
+  // DeleteDatabaseService removes the specified DatabaseService.
+  rpc DeleteDatabaseService(types.ResourceRequest) returns (google.protobuf.Empty);
+  // DeleteAllDatabaseServices removes all registered DatabaseServices.
```

Comment:
It depends on the `Backend.DeleteRange` implementation of the backend we are using.
It looks like an error can occur and we only delete a subset of the resources, in most implementations.

This method is meant to be used in cache, which usually has an in-memory backend.
That backend should be more resilient than the others (no disk or network interactions), unless something really bad (OOM, SIGKILL, ?) is happening.

Reply:
> What do you mean "meant to be used in cache"? It won't cause deletion from the underlying storage?

It will, but it's usage is meant to be only in cache
Something like this:
https://github.com/gravitational/teleport/blob/028d809e75296161cef92ac40e7469239bb513ba/lib/cache/collections.go#L1281

Added the lack of guarantees it gives us
It really depends on the backend we are using.

Reply:
Ack, thanks for the docs.